src/cwe_checker_lib/src/analysis/taint/state.rs

```diff
@@ -411,6 +404,11 @@ impl State {
     pub fn is_empty(&self) -> bool {
         self.memory_taint.is_empty() && self.register_taint.is_empty()
     }
+
+    /// Check whether there is any tainted memory in the state.
+    pub fn has_memory_taint(&self) -> bool {
+        !self.memory_taint.is_empty()
+    }
```

Comment:
There is again an annoying corner case: A quick glance at `State::save_taint_to_memory` suggests that we may have the case that we overwrite all taint in a memory object, but do not delete the corresponding `MemRegion` object. Then `self.memory_taint.is_empty()` would return `false`, because we still track an object, despite it containing no taint. And that may trigger panics in `update_return` (if users implement `update_return_callee` without deleting such objects). On the other hand, one could still consider this case an error on the trait implementer side, so it is not a merge blocker for this PR.

What do you think? Should we catch this corner case or do we consider it a responsibility for the implementer?